server/events/locking/locking.go

```diff
@@ -66,6 +98,49 @@ func NewClient(backend Backend) *Client {
 	}
 }
 
+// LockApply acquires global apply lock.
+func (c *Client) LockApply() (ApplyCommandLockResponse, error) {
+	response := ApplyCommandLockResponse{}
+
+	applyCmdLock, err := c.backend.LockCommand(models.ApplyCommand, time.Now().Local())
+	if err != nil {
+		return response, err
+	}
+
+	if applyCmdLock != nil {
+		response.Present = true
+		response.Time = applyCmdLock.Time
+	}
+	return response, nil
+}
+
+// UnlockApply releases a global apply lock.
+func (c *Client) UnlockApply() error {
+	err := c.backend.UnlockCommand(models.ApplyCommand)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// CheckApplyLock retrieves an apply command lock if present.
+func (c *Client) CheckApplyLock() (ApplyCommandLockResponse, error) {
```

Comment:
Sounds to me like this should also follow the same logic of checking the server configuration as well. We should also probably not allow unlocking of atlantis if it's configured to disable applies through server configuration both.

Right now the implementation doesn't do that which is good, however, it'll say it's locked and unlocked it, but applies still won't work if the server configuration has it defined.